commitment_test.go

```diff
@@ -0,0 +1,159 @@
+package raft
+
+import (
+	"testing"
+)
+
+// Returns a slice of server names of size n.
+func voters(n int) []string {
+	if n > 7 {
+		panic("only up to 7 servers implemented")
+	}
+	return []string{"s1", "s2", "s3", "s4", "s5", "s6", "s7"}[:n+1]
```

Comment:
Nice catch, as this off-by-one masked a critical off-by-one in the code under test.